blueoil/networks/classification/quantize_example.py

```diff
@@ -0,0 +1,134 @@
+        super().__init__(*args, **kwargs)
+
+        self.activation = lambda x: tf.nn.leaky_relu(x, alpha=0.1, name="leaky_relu")
+
+    def base(self, images, is_training):
+        assert self.data_format == "NHWC"
+        channel_data_format = "channels_last"
+
+        self.inputs = self.images = images
+
+        with tf.compat.v1.variable_scope("block_1"):
+            conv = conv2d("conv", self.inputs, filters=32, kernel_size=3,
+                          activation=None, use_bias=False, data_format=channel_data_format,
+                          kernel_initializer=tf.contrib.layers.variance_scaling_initializer())
+            batch_normed = batch_norm("bn", conv, is_training=is_training, decay=0.99, scale=True, center=True,
+                                      data_format=self.data_format)
+            self.block_1 = self.activation(batch_normed)
+
+        self.block_last = conv2d("block_last", self.block_1, filters=self.num_classes, kernel_size=1,
+                                 activation=None, use_bias=True, is_debug=self.is_debug,
+                                 kernel_initializer=tf.random_normal_initializer(mean=0.0, stddev=0.01),
+                                 data_format=channel_data_format)
+
+        h = self.block_last.get_shape()[1].value
+        w = self.block_last.get_shape()[2].value
+        self.pool = tf.layers.average_pooling2d(name='global_average_pool', inputs=self.block_last,
+                                                pool_size=[h, w], padding='VALID', strides=1,
+                                                data_format=channel_data_format)
+        self.base_output = tf.reshape(self.pool, [-1, self.num_classes], name="pool_reshape")
+
+        return self.base_output
+
+
+class SampleNetworkQuantize(SampleNetwork):
+    """Quantize Sample Network."""
+
+    def __init__(
+            self,
+            quantize_first_convolution=True,
+            quantize_last_convolution=True,
+            activation_quantizer=None,
+            activation_quantizer_kwargs=None,
+            weight_quantizer=None,
+            weight_quantizer_kwargs=None,
+            *args,
+            **kwargs
+    ):
+        """
+        Args:
+            quantize_first_convolution(bool): use quantization in first conv.
+            quantize_last_convolution(bool): use quantization in last conv.
+            weight_quantizer (callable): weight quantizer.
+            weight_quantizer_kwargs(dict): Initialize kwargs for weight quantizer.
+            activation_quantizer (callable): activation quantizer
+            activation_quantizer_kwargs(dict): Initialize kwargs for activation quantizer.
+        """
+
+        super().__init__(*args, **kwargs)
+
+        self.quantize_first_convolution = quantize_first_convolution
+        self.quantize_last_convolution = quantize_last_convolution
+
+        activation_quantizer_kwargs = activation_quantizer_kwargs if activation_quantizer_kwargs is not None else {}
```

Comment:
I think we don't need this `a if a is not None else {}` trick at all here.  This is only necessary when we modify given args.
Here, we don't modify `activation_quantizer_kwargs` thus we can simply use `{}` as the default arg.